### PR TITLE
Add linting rules

### PR DESCRIPTION
**Pull request by steam**
Uses the linting rules from https://github.com/ello/webapp

Did I miss anything @mkitt / @666?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,17 +16,23 @@
       "transform-flow-strip-types"
     ]
   },
+  "eslintConfig": {
+    "extends": "ello"
+  },
   "scripts": {
     "build": "babel src --out-dir ./",
     "build:watch": "babel src --out-dir ./ --watch"
   },
```
Comment (mkitt) [inline package.json]: We should add:
```
    "lint": "eslint src test"
```

File: package.json [hunk 2]
```diff
@@ -16,18 +16,54 @@
...
+          "ignore": [
+            "calc",
+            "flexbox",
+            "viewport-units",
+            "will-change"
+          ]
+        }
+      ]
+    }
+  },
```
Comment (mkitt) [inline package.json]: We don't need any of the style lint stuff in here. This repo shouldn't ever have CSS.

File: package.json [hunk 3]
```diff
@@ -16,18 +16,54 @@
...
   "scripts": {
     "build": "babel src --out-dir ./",
-    "build:watch": "babel src --out-dir ./ --watch"
+    "build:watch": "babel src --out-dir ./ --watch",
+    "flow": "flow",
+    "lint:css": "stylelint 'src'",
```
Comment (mkitt) [inline package.json]: You can kill this.

File: package.json [hunk 4]
```diff
@@ -16,18 +16,54 @@
...
+    "lint:js": "eslint src test",
+    "lint": "yarn run lint:js && yarn run lint:css"
```
Comment (mkitt) [inline package.json]: probably just use the `lint` target only here... no need for `lint:js`

File: package.json [hunk 5]
```diff
@@ -16,18 +16,54 @@
...
     "babel-plugin-transform-flow-strip-types": "6.22.0",
     "babel-preset-es2015": "6.24.1",
     "babel-preset-react": "6.24.1",
     "babel-preset-stage-0": "6.24.1",
-    "flow-bin": "0.45.0"
+    "eslint": "3.19.0",
+    "eslint-config-ello": "1.0.3",
+    "flow-bin": "0.45.0",
+    "stylelint": "7.10.1",
+    "stylelint-config-primer": "1.4.0"
```
Comment (mkitt) [inline package.json]: Kill these 2 style lint packages

**Review by mkitt — COMMENTED** [review]
Probably want to add a `.flowconfig` file to.

File: package.json (hunk 2 above)
Comment (steam) [reply package.json]: Cool, I'll kill them. Is this outdated as it refers to non js based css styles and we're moving away from them?

**Comment by steam** [comment]
Ok @mkitt, stylelint be-gone.

File: .flowconfig [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[ignore]
+.*/node_modules/stylelint/.*
```
Comment (mkitt) [inline .flowconfig]: You can kill this line.

File: .flowconfig [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+.*/node_modules/fbjs/.*
+.*/node_modules/nock/.*
```
Comment (mkitt) [inline .flowconfig]: I think you can kill this line too. (`nock`)

**Review by mkitt — APPROVED** [review]
Couple of other minor removals but I think it's good to go after that.

File: .flowconfig (hunk 6 above)
Comment (steam) [reply .flowconfig]: Good catch.

File: .flowconfig (hunk 7 above)
Comment (steam) [reply .flowconfig]: Cool, I'll axe them both.

File: package.json [hunk 8]
```diff
@@ -16,17 +16,29 @@
       "transform-flow-strip-types"
     ]
   },
+  "eslintConfig": {
+    "extends": "ello",
+    "globals": {
+      "AndroidInterface": false,
+      "ENV": false
```
Comment (rynbyjn) [inline package.json]: I would image we don't want either of these globals in brains. `AndroidInterface` should probably only live in webapp and `ENV` shouldn't be accessed via a 3rd party package.

File: package.json [hunk 9]
```diff
@@ -16,17 +16,29 @@
...
+    }
+  },
   "scripts": {
     "build": "babel src --out-dir ./",
-    "build:watch": "babel src --out-dir ./ --watch"
+    "build:watch": "babel src --out-dir ./ --watch",
+    "flow": "flow",
+    "lint": "eslint src test"
```
Comment (rynbyjn) [inline package.json:26]: I assume this doesn't blow up even without the `test` directory yet?

**Review by rynbyjn — CHANGES_REQUESTED** [review]
I would just kill the `globals` part of the `eslintConfig` and this is GTG 🤘

File: package.json (hunk 8 above)
Comment (steam) [reply package.json]: Yeah, this was added in an effort to get `eslint` to run by itself this morning when Matt and I were troubleshooting it. It didn't help one way or another so yeah, lets kill it.

File: package.json (hunk 9 above)
Comment (steam) [reply package.json:26]: It doesn't. And the test directory is coming later today I hope. Different PR tho.

**Review by rynbyjn — APPROVED** [review]
Word 🤘

**Pull request merged**